protocol_rfcs/variant-type.md

```diff
@@ -0,0 +1,87 @@
+# Variant Data Type
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/2864**
```

Comment:
Would be better to follow the structure of the protocol doc - Requirements for readers and requirements for writers.